File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.py
class Solution(object):
    def longestOnes(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """

        left = 0
        zeros = 0
        best = 0
     
        for right in range(len(nums)):
            if (nums[right] == 0):
                zeros += 1
            while (zeros > k):
                
                if (nums[left] == 0):
                    zeros -= 1
                left += 1
            best = max(best, right - left + 1)
           
         
                
        
        
        return best
                
            

             
        

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        // if (k == 0) {
        //     int best = 0;
        //     int count = 0;
        //     int i = 0;
        //         while(i < nums.length) {
                
        //         while (nums[i] == 1 && i < nums.length - 1) {
        //             count++;
        //             i++;
        //         }
        //         best = Math.max(count, best);
        //         i++;
        //         }
        //     return best;        
        // }
        
        int zeros = 0;
        int left = 0;
        int best = 0;
        for (int right = 0; right < nums.length; right++) {
            if (nums[right] == 0) zeros++;
        
            while (zeros > k) {
                if (nums[left] == 0) zeros--;
                left++;
            }
          
            best = Math.max(best, right - left + 1);
           
        }
        return best;
    }
}